parser-typechecker/src/Unison/Util/Bytes.hs

```diff
@@ -1,42 +1,78 @@
 {-# Language ViewPatterns #-}
+{-# Language GeneralizedNewtypeDeriving #-}
+{-# Language BangPatterns #-}
 {-# OPTIONS_GHC -fno-warn-orphans #-}
 
 module Unison.Util.Bytes where
 
 import Control.DeepSeq (NFData(..))
 import Data.Bits (shiftR, shiftL, (.|.))
 import Data.Char
-import Data.Memory.PtrMethods (memCompare, memEqual)
-import Data.Monoid (Sum(..))
-import Foreign.Ptr (Ptr, plusPtr)
-import Foreign.Storable (poke)
-import System.IO.Unsafe (unsafeDupablePerformIO)
 import Unison.Prelude hiding (ByteString, empty)
-import Basement.Block (Block)
-import qualified Data.ByteString.Lazy as LB
-import qualified Data.ByteArray as B
+import qualified Data.ByteString as B
+import qualified Data.ByteArray as BA
 import qualified Data.ByteArray.Encoding as BE
-import qualified Data.FingerTree as T
+import qualified Data.ByteString.Lazy as LB
 import qualified Data.Text as Text
+import qualified Unison.Util.Rope as R
+import qualified Data.Vector.Primitive as V
+import qualified Data.Vector.Storable as SV
+import qualified Data.Vector.Storable.ByteString as BSV
+import Foreign.Storable (pokeByteOff)
 import qualified Codec.Compression.Zlib as Zlib
 import qualified Codec.Compression.GZip as GZip
 
--- Block is just `newtype Block a = Block ByteArray#`
-type ByteString = Block Word8
+type ByteString = V.Vector Word8
```

Comment:
Maybe call this `ShortByteString` or something to help imply when you would use this instead of Bytes?
Or is this not meant to be exported, in which case maybe add an export list?
If I just saw `ByteString` and `Bytes` in a file it's not clear which is used for what 😄

Reply:
Went with `Chunk` as the name.